components/odh-notebook-controller/controllers/notebook_runtime.go

```diff
@@ -0,0 +1,248 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/go-logr/logr"
+	nbv1 "github.com/kubeflow/kubeflow/components/notebook-controller/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+const configMapName = "pipeline-runtime-images"
+const mountPath = "/opt/app-root/pipeline-runtimes/"
+const volumeName = "runtime-images"
+
+// checkConfigMapExists verifies if a ConfigMap exists in the namespace.
+func (r *OpenshiftNotebookReconciler) checkConfigMapExists(ctx context.Context, configMapName, namespace string) (bool, error) {
+	configMap := &corev1.ConfigMap{}
+	err := r.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: namespace}, configMap)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return false, nil // ConfigMap not found
+		}
+		return false, err // Some other error occurred
+	}
+	//r.Log.Info("ConfigMap found", "ConfigMap.Name", configMapName, "Namespace", namespace)
+	return true, nil // ConfigMap exists
+}
+
+func (r *OpenshiftNotebookReconciler) syncRuntimeImagesConfigMap(ctx context.Context, notebookNamespace string) error {
+	log := r.Log.WithValues("namespace", notebookNamespace)
+
+	// Create a dynamic client
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Error(err, "Error creating cluster config")
+		return err
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Error(err, "Error creating dynamic client")
+		return err
+	}
```

Comment:
```suggestion
	// Create a dynamic client
	dynamicClient, err := dynamic.NewForConfig(r.Config)
	if err != nil {
		log.Error(err, "Error creating dynamic client")
		return err
	}
```

Reply:
> panic: runtime error: invalid memory address or nil pointer dereference

because you also need to initialize `r.Config`

Reply:
Right, I fixed that